### ISSUE TITLE
Filter events in CMS

### ISSUE THREAD

**Issue by saumier** [culturecreates/footlight-app#74]
As a user (any role) I can filter by published state, user.  The user filter is essential to guests who want to view only their own events. The published state filter is essential to admins who have 300 events, and want to see which events are ready for review. Other filters are moved out of MVP.

**Comment by saumier** [comment]
@AbhishekPAnil You may take up this issue now that @SuhailAliyar has deployed the need API changes.

**Comment by SuhailAliyar** [comment]
@AbhishekPAnil The filter is updated so that a list of publish states can now be passed to filter events in CMS.

**Comment by saumier** [comment]
@troughc Please test in staging or production as both are updated with this version. Thx.

**Comment by troughc** [comment]
Tested in Chrome using the tout culture 

Three issues related to the User Filter: 
First issue:
The list of User names in the User filter capitalizes the first and second name. However, in the event list, this name is not capitalized. 3 examples:  gfinch in event list and GFinch in user filter; sadmin in event list, SAdmin in User filter; ctroughton in event list CTroughton in user filter. Are these the same user? If yes, the name should appear exactly the same (lower case). Is this possible?

Second issue/question:
Can the user names be alphabetical? Right now they are in the order in which they were added to the system. 
For example, the order could either be based on:
- first letter of first name: ctroughton, gfinch, sadmin
- first letter of last name: sadmin, gfinch, ctroughton
@saumier what would be better?

**Comment by saumier** [comment]
@troughc I propose the user names be displayed first letter of first name.  Please assign to Abhishek once you have made your decision.

**Comment by troughc** [comment]
Agreed. Please list usernames alphabetically @AbhishekPAnil  according to the first letter of the first name. Also see my comment above with other issue regarding capitalization.

**Comment by troughc** [comment]
Two more thing:s the order of filters is wrong according to the figma - Publication comes before users.
Can you also check the spacing within the filters - for example, when nothting is selected in the filter, the word is not centred. Thank you.

**Comment by troughc** [comment]
@AbhishekPAnil, the 'Users' in the french version of the UI is in English, It should be "Utilisateurs"

**Comment by troughc** [comment]
Tested

Search field needs to be improved later but it works as requested for mvp.

There is an issue with the User filter that we need to address:

What I type does not match the results I get/ what I see. For example, 
- I type greg and I get gfinch options (there are no user names that show 'greg') - maybe this is okay because greg is the first name of a user.
- I type gf I get ONE gfinch option but i know there are four usernames that have gfinch; 
- I type g, I get many options

So, the search term I enter isn't looking for what is being displayed (the usernames). Rather, it seems to be searching first name, last name (though I am not sure). 
Since we are listing usernames, then we should be able to search for those usernames. If I search for ctroughton, gfinch, ap anil, finch, ap, anil etc. (all or part of the username), I should find them (I don't). 

And maybe, if I search for the first name of a user (because I know it), I would also find the associated username, that would be okay.

see screenshots for different results

![Image](https://user-images.githubusercontent.com/64489312/219510735-31227f65-4bf7-49f0-b2c8-623a3e67c27f.png)



![Image](https://user-images.githubusercontent.com/64489312/219510746-43831932-e81f-4d5b-b099-7b149c0a6cb2.png)



![Image](https://user-images.githubusercontent.com/64489312/219510778-b688a79e-33e5-440d-8911-949079c5e8b3.png)



![Image](https://user-images.githubusercontent.com/64489312/219509785-732db1ae-4e96-4bbe-b7f9-59c06a2dba06.png)



![Image](https://user-images.githubusercontent.com/64489312/219509810-87a1eb89-bef0-4c4f-a33b-8ae6da6d0d51.png)

**Comment by troughc** [comment]
Tested: PASSED

**Comment by troughc** [comment]
Found a new one @AbhishekPAnil.
1. Click the users button.
2. You can either type a name  the field  or scroll to a user, for example finch. Select one of Gregory's profiles.  count goes to 1
3. Select 'my events', and any other users you want (either by typing or scrolling count goes up and down as expected (GOOD)
4. Click Clear button. The filter is cleared and the dropdown is closed.
5. open the users button - nothing is selected. There are two possibilities:
FIRST: if you typed soemthing before, it appears in the field. and the dropdown displays options according to your text
SECOND: If you did not type anything in the field, the field appears to be empty. So the filter looks like it is in its default state.
BOTH of these situations result in the following:
6. select any option from the dropdown - all your previous selections are reselected.

I took the screenshot below after doing all these steps - in step 6 I selected one profile (and 6 others that I had added from the previous search were also added - count went to 7 instead of 1)



![Image](https://user-images.githubusercontent.com/64489312/220201781-4d5eebcf-9d0e-4e26-82aa-6c47e10f6e22.png)

**Comment by AbhishekPAnil** [comment]
@troughc Have fixed the above issue now and deployed in staging.Please test and let me know.Thank you.
